crypto/s2n_cipher.h

```diff
@@ -26,8 +26,15 @@
 
 #include "utils/s2n_blob.h"
 
+#if defined(OPENSSL_IS_BORINGSSL) || defined(OPENSSL_IS_AWSLC)
```

Comment:
Another philosophical question: should this be powered by the existence of `EVP_AEAD_CTX` directly?

Reply:
I think that would be a bad idea. `OPENSSL_IS_BORINGSSL` [boringssl](https://github.com/google/boringssl/blob/master/PORTING.md#porting-from-openssl-to-boringssl) and `OPENSSL_IS_AWSLC` [aws-lc](https://github.com/awslabs/aws-lc/blob/main/PORTING.md#porting-from-opensslboringssl-to-aws-lc) are the recommended ways to distinguish from other libcrypo's